.github/actions/check-trivyignore/action.yaml

```
@@ -0,0 +1,39 @@
+name: Check Trivyignore
+description: 'Check if the trivyignore file is up to date'
+# This action assumes that trivy is already set up in the previous steps
+# Reference: https://github.com/canonical/operator-workflows/blob/c68a31ba2c953765654b2c04b6a760b65f9bfe83/.github/workflows/integration_test.yaml#L305
+
+inputs:
+  image-ref:
+    description: 'The image to be scanned'
+    required: true
+  trivyignores:
+    description: 'The path to the trivyignore file'
+    required: true
+  severity:
+    description: 'The severity level to be scanned'
+    default: 'HIGH,CRITICAL'
+  skip-files:
+    description: 'The files to be skipped'
+    default: '/bin/pebble,/usr/bin/pebble'
+
+runs:
+  using: "composite" 
+  steps:
+    - name: "Check unnecessary trivyignore entries"
+      shell: bash
+      run: |
+        if [ -f "${{ inputs.trivyignores }}" ]
+          then
+            output=$(trivy image ${{ inputs.image-ref }} --severity ${{ inputs.severity }} -q -f json --ignorefile "" --skip-files ${{ inputs.skip-files }} \
+                      | jq -r '.Results[].Vulnerabilities[].VulnerabilityID' 2>/dev/null || echo "No vulnerabilities found")
+            line=0
+            while read CVE;
+            do
+              line=$(( line + 1 ))
+              if [[ "$output" != *"$CVE"* ]] && [[ ! "$CVE" =~ ^#.* ]]
```

nit: with new test you can put the operators inside. 
```suggestion
              if [[ "$output" != *"$CVE"* && ! "$CVE" =~ ^#.* ]]
```

---

We could also add a garud that discards lines that do not fit CVE-\d+-\d+